README.md

```diff
@@ -35,6 +35,27 @@ $ docker build -t my-nodejs-app .
 $ docker run -it --rm --name my-running-app my-nodejs-app
 ```
 
+If you prefer Docker Compose:
+
+```yaml
+version: "2"
+services:
+  node:
+    image: "node:4-onbuild"
```

Comment:
As per the [documentation](https://docs.docker.com/compose/reference/up/), Docker Compose does build when run through Docker Compose:

> "Builds, (re)creates, starts, and attaches to containers for a service"

Using the `-onbuild` tag is awesome for Docker Compose precisely because you don't need the Dockerfile. 

Try this: create only the following files in a folder, then run `docker-compose up -d`. You will see 'Hello World' in 0.0.0.0:8080.

File: **docker-compose.yaml**

```yaml
version: "2"
services:
  node:
    image: "node:4-onbuild"
    environment:
      - NODE_ENV=production
    volumes:
      - ./:/usr/src/app
    ports:
      - "8080:8080"
```


File: **package.json**
```json
{
    "name": "docker-test",
    "version": "1.0.0",
    "description": "",
    "main": "index.js",
    "scripts": {
        "test": "echo \"Error: no test specified\" && exit 1",
        "start": "node index.js"
    },
    "keywords": [],
    "author": "",
    "license": "ISC",
    "dependencies": {
        "express": "^4.14.1"
    }
}

```


File: **index.js**
```js
var express = require('express')
var app = express()

app.get('/', function (req, res) {
  res.send('Hello World!')
})

app.listen(8080, function () {
  console.log('Example app listening on port 8080!')
})
```

File: **.dockerignore** 
```
node_modules
```